routes/eth.go

```diff
@@ -367,14 +375,113 @@ func (fes *APIServer) QueryETHRPC(ww http.ResponseWriter, req *http.Request) {
 	}
 }
 
+type MetamaskSignInRequest struct {
+	AmountNanos uint64
+	Signer      []byte
+	Message     []byte
+	Signature   []byte
+}
+type MetamaskSignInResponse struct {
+	TxnHash *lib.BlockHash
+}
+
+func (fes *APIServer) MetamaskSignIn(ww http.ResponseWriter, req *http.Request) {
+	// Give the user starter deso if this is their first time signing in with through metamask and if they don't have Deso
+	DEFAULT_ERROR := "MetamaskSignin: something went wrong with processing your airdrop "
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	// Validate the  request object
+	requestData := MetamaskSignInRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("MetamaskSignin: Problem parsing request body: %v", err))
+		return
+	}
+	// get the deso public address of the user
+	recipientPublicKey := lib.Base58CheckEncode(requestData.Signer, false, fes.Params)
+	recipientBytePK := []byte(recipientPublicKey)
+	// get the public eth address of the user
+	recipientEthAddress, err := publicKeyToEthAddress(requestData.Signer)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf(DEFAULT_ERROR, err))
+	}
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf(DEFAULT_ERROR, err))
+	}
```

Comment:
```suggestion
	if err != nil {
		_AddBadRequestError(ww, fmt.Sprintf(DEFAULT_ERROR, err))
		return
	}
```
we need to return after hitting an error.